sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -403,9 +403,44 @@ std::unique_ptr<RawResponse> CurlTransport::Send(Request& request, Context const
   auto response = session->ExtractResponse();
   // Move the ownership of the CurlSession (bodyStream) to the response
   response->SetBodyStream(std::move(session));
+  // check the consistency of the keep alive headers between the request and the response
+  ValidateKeepAliveHeaders(request, response);
+
   return response;
 }
 
+void CurlTransport::ValidateKeepAliveHeaders(
+    Request& request,
+    std::unique_ptr<RawResponse>& response)
+{
+  // if the server supports keep alive the headers should be present in the response. If they are
+  // they should be the same as the request headers.
+  if (response->GetHeaders().find("Connection") != response->GetHeaders().end()
+      && request.GetHeader("Connection").HasValue()
+      && response->GetHeaders().find("Keep-Alive") != response->GetHeaders().end()
```

Comment:
Does the RFC require that the request contain a Connection and Keep-Alive header?

Reply:
yes both need to be present. and according to the info i found if the server agrees to the keep alive it will send back the same values. thus if they are not present on the response they i assume the server either does not support them or does not agree, in any case i remove them from the request and connection options in order tto not confuse the cache any further when the connection is released back to the pool